modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -947,6 +1027,24 @@ void ONNXImporter::populateNet(Net dstNet)
             else
                 layerParams.type = "Identity";
         }
+        else if (layer_type == "ConstantOfShape")
+        {
+            Mat input = getBlob(node_proto, constBlobs, 0);
+            CV_CheckEQ(input.dims, 2, "");
+            CV_CheckEQ(input.size[1], 1, "");
+            float fill_value = layerParams.blobs.empty() ? 0 : layerParams.blobs[0].at<float>(0, 0);
+            std::vector<int> sizes;
+            for (int i = 0; i < input.total(); i++)
+            {
+                int size = input.at<int>(i, 0);
+                CV_Assert(size > 0);
+                sizes.push_back(size);
+            }
+            Mat tensor(sizes, CV_32F, fill_value);
```

Comment:
```cpp
MatShape inpShape = getBlob(node_proto, constBlobs, 0);
Mat tensor(inpShape.size(), &inpShape[0], CV_32F, Scalar(fill_value));
```

Reply:
In this PR I check values in constBlob

Reply:
Got it, thanks!
So, maybe we can cast it to `MatShape` and check values of vector rather than values in Mat?

```cpp
MatShape inpShape = getBlob(node_proto, constBlobs, 0);
for (int i = 0; i < inpShape.size(); i++)
    CV_CheckLT(inpShape[i], 0, "");
```